testsuite/tests/c-api/alloc_async_stubs.c

```diff
@@ -12,9 +12,9 @@ value stub(value ref)
 
   printf("C, before: %d\n", Int_val(Field(ref, 0)));
 
-  /* First, do enough major allocation to trigger a major collection */
+  /* First, do enough major allocations to do a full major collection cycle */
   coll_before = Caml_state_field(stat_major_collections);
-  while (Caml_state_field(stat_major_collections) == coll_before) {
+  while (Caml_state_field(stat_major_collections) <= coll_before+1) {
```

Comment:
I'm going to trust you that this change is related to the other changes.

Reply:
It is related in the sense that the changes in this PR triggered a bug in the test.

But I think this is only